DNN Platform/Library/Common/Globals.cs

```diff
@@ -2898,10 +2901,10 @@ public static string UserProfileURL(int userId)
         /// </summary>
         /// <returns>Formatted URL.</returns>
         [Browsable(false), DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)]
+        [Obsolete("Deprecated in Platform 9.4.2. Scheduled removal in v11.0.0.")]
         public static string NavigateURL()
         {
-            PortalSettings portalSettings = PortalController.Instance.GetCurrentPortalSettings();
-            return NavigateURL(portalSettings.ActiveTab.TabID, Null.NullString);
+            return DependencyProvider.GetService<INavigationManager>().NavigateURL();

```

Comment:
This makes sense to me, we can update this to use `GetRequiredService` which will throw an exception if it can't resolve `INavigationManager`.

The less chance of a `NullReferenceException` the better

Reply:
I've pushed b669f8dc608acbfc19c640cf4eff0bdf8f38d26e with this change